Source/ResponseSerialization.swift

```diff
@@ -22,9 +22,100 @@
 
 import Foundation
 
-// MARK: String
+// MARK: - ResponseSerializer
+
+/**
+    The type in which all response serializers must conform to in order to serialize a response.
+*/
+public protocol ResponseSerializer {
+    /// The type of serialized object to be created by this `ResponseSerializer`.
+    typealias SerializedObject
+
+    /// A closure used by response handlers that takes a request, response, and data and returns a serialized object and any error that occured in the process.
+    var serializeResponse: (NSURLRequest?, NSHTTPURLResponse?, NSData?) -> (SerializedObject?, NSError?) { get }
+}
+
+/**
+    A generic `ResponseSerializer` used to serialize a request, response, and data into a serialized object.
+*/
+public struct GenericResponseSerializer<T>: ResponseSerializer {
+    /// The type of serialized object to be created by this `ResponseSerializer`.
+    public typealias SerializedObject = T
+
+    /// A closure used by response handlers that takes a request, response, and data and returns a serialized object and any error that occured in the process.
+    public var serializeResponse: (NSURLRequest?, NSHTTPURLResponse?, NSData?) -> (SerializedObject?, NSError?)
+}
+
+// MARK: - Default
+
+extension Request {
+
+    /**
+        Adds a handler to be called once the request has finished.
+
+        :param: queue The queue on which the completion handler is dispatched.
+        :param: responseSerializer The response serializer responsible for serializing the request, response, and data.
+        :param: completionHandler The code to be executed once the request has finished.
+
+        :returns: The request.
+    */
+    public func response<T: ResponseSerializer, V where T.SerializedObject == V>(
+        queue: dispatch_queue_t? = nil,
+        responseSerializer: T,
```

Comment:
These are all great points @NicholasTD07 that I certainly considered. 

I did everything possible here to keep the public APIs as consistent as possible, while still fixing the root problem of always using an `AnyObject?` return type in the root `response` method. There are technically three changes that break semver.

* The response serializer implementations have been changed from `class` to `static` methods.
* The `serializer` parameter in the `response` method has been renamed to `responseSerializer` to provide more clarity.
* The type on `responseSerializer` is no longer a `Serializer` closure, but is instead a generic that conforms to the `ResponseSerializer` protocol.

Generally, I'd fall into the camp that this is an Alamofire 2.0 type of change. However, these changes above will affect only a small group of consumers. The majority of users are not ever going to implement the `response` method directly or ever subclass any of the `ResponseSerializer` methods. Additionally, the Alamofire 1.3.0 release is the last release that we're planning before Alamofire 2.0 work begins. The major downside to deferring this change to 2.0 is that this change wouldn't drop for months. The Alamofire 2.0 release is currently slated to go out roughly with the Swift 2.0 GM. This could still change if we decide to actually push a release on a beta version of Xcode, but I doubt that we'll go that route.

Therefore, to me, I feel that the slight deviation from semver here to serve the greater overall good is the right way to go. Renaming a variable for a very low API that gives me way more power and safety is all good in my book. But that's me, and I want to make sure that the community and the TC agrees. If the everyone would rather follow semver to the book here, I'm happy to defer this change until the 2.0 release.